dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -83,53 +103,80 @@ def __add__(self, other: CategoricalColumn) -> CategoricalColumn:
+                if profile1_product < profile2_product:
+                    merged_profile.max_sample_size_to_check_stop_condition = (
+                        self.max_sample_size_to_check_stop_condition
+                    )
+                    merged_profile.stop_condition_unique_value_ratio = (
+                        self.stop_condition_unique_value_ratio
+                    )
+                else:
+                    merged_profile.stop_condition_unique_value_ratio = (
+                        other.stop_condition_unique_value_ratio
+                    )
+                    merged_profile.max_sample_size_to_check_stop_condition = (
+                        other.max_sample_size_to_check_stop_condition
+                    )
+
+                # Check merged profile w/ stop condition
+                if merged_profile._check_stop_condition_is_met(
+                    merged_profile.sample_size, merged_profile.unique_ratio
+                ):
+                    merged_profile._stopped_at_unique_ratio = (
+                        merged_profile.unique_ratio
+                    )
+                    merged_profile._stopped_at_unique_count = (
+                        merged_profile.unique_count
+                    )
+                    merged_profile._categories = {}
+                    merged_profile._stop_condition_is_met = True
+
+            else:
+                if self.sample_size > other.sample_size:
+                    merged_profile._stopped_at_unique_ratio = self.unique_ratio
+                    merged_profile._stopped_at_unique_count = self.unique_count
+                    merged_profile.sample_size = self.sample_size
+                else:
+                    merged_profile._stopped_at_unique_ratio = other.unique_ratio
+                    merged_profile._stopped_at_unique_count = other.unique_count
+                    merged_profile.sample_size = other.sample_size
+
+                # If either profile has hit stop condition, remove categories dict
                 merged_profile._categories = {}
                 merged_profile._stop_condition_is_met = True
 
         else:
-            if self.sample_size > other.sample_size:
-                merged_profile._stopped_at_unique_ratio = self.unique_ratio
-                merged_profile._stopped_at_unique_count = self.unique_count
-                merged_profile.sample_size = self.sample_size
-            else:
-                merged_profile._stopped_at_unique_ratio = other.unique_ratio
-                merged_profile._stopped_at_unique_count = other.unique_count
-                merged_profile.sample_size = other.sample_size
-
-            # If either profile has hit stop condition, remove categories dict
-            merged_profile._categories = {}
-            merged_profile._stop_condition_is_met = True
+            raise Exception(
+                "Unable to add two profiles is only one is using count min sketch."
```

Comment:
just a grammar fix here:

> "Unable to add two profiles: One is using count min sketch and the other is using full."

Reply:
We need a test here to validate this exception.

Reply:
added